Sources/NIO/CircularBuffer.swift

```diff
@@ -227,7 +227,7 @@ extension CircularBuffer: RandomAccessCollection {
     /// Returns the index offset by `distance` from `index`.
     @inlinable
     public func index(_ i: Index, offsetBy distance: Int) -> Index {
-        return .init(backingIndex: (i.backingIndex + distance) & self.mask,
+        return .init(backingIndex: (i.backingIndex &+ distance) & self.mask,
```

Comment:
I don't think this one can be left unchecked since `distance` be arbitrarily large. cc @weissi

Reply:
@glbrntt I _think_ it's undefined behaviour if you ask for an OOB index, let me check the docs.

Reply:
@glbrntt yes, quoting the [docs](https://developer.apple.com/documentation/swift/bidirectionalcollection/3024450-index)

> The value passed as distance must not offset i beyond the bounds of the collection.

Reply:
so I think we're fine here.

Reply:
ah okay, sounds good to me